api/include/opentelemetry/logs/logger.h

```diff
@@ -55,69 +51,92 @@ class Logger
    * @throws No exceptions under any circumstances.
    */
 
-  /* The below method is a logging statement that takes in a LogRecord.
-   * A default LogRecord that will be assigned if no parameters are passed to Logger's .log() method
-   * which should at minimum assign the trace_id, span_id, and timestamp
+  /**
+   * Logs a LogRecord, which contains all the fields of the Log Data Model. Normally called
+   * indirectly from other log() Methods, but can be called directly for high detail.
```

Comment:
I think we are going to rename the method name from `log()` to `Log()`. So make sure the new comment matches the new name.

Reply:
Once #422 is merged, it should ideally overwrite all the changes in this PR from `logger.h` and `log_record.h`. I included a subset of the changes from the other PR here so the CI would run properly. Currently since the log() method is named with a lowercase, I think it would be good to keep the comment consistent here.

Reply:
Currently since the log() method is named with a lowercase, I think it would be good to keep the comment consistent with the code. This change could be part of #422?